azkaban-spi/src/main/java/azkaban/spi/AzkabanEventReporter.java

```diff
@@ -0,0 +1,19 @@
+package azkaban.spi;
+
+import java.util.Map;
+
+/**
+ * Implement this interface to report flow and job events. Event reporter
+ * can be turned on by setting the property 'azkaban.event.reporting.enabled' to true.
+ *
+ * By default, a KafkaAvroEventReporter is provided. Alternate implementations
+ * can be provided by setting the property 'azkaban.event.reporting.class'.
+ *
+ * The constructor will be called with a azkaban.utils.Props object passed as
+ * the only parameter. If such a constructor doesn't exist, then the AzkabanEventReporter
+ * instantiation will fail.
+ */
+public interface AzkabanEventReporter {
+
+  boolean report(IEvent event, Map<String, String> metadata);
```

Comment:
Why is it better than use EventType directly in this interface?

Reply:
Changed to use EventType directly.